src/conn_interface.c

```diff
@@ -131,14 +131,17 @@ void createSocketConnection(void (* initKeypress)())
         struct timespec start, stop, diff;
         int time_taken_ms;
 
-        clock_gettime(CLOCK_REALTIME, &start);
+        clock_gettime(CLOCK_MONOTONIC, &start);
         nopoll_loop_wait(ctx, 5000000);
-        clock_gettime(CLOCK_REALTIME, &stop);
+        clock_gettime(CLOCK_MONOTONIC, &stop);
 
         timespec_diff(&start, &stop, &diff);
         time_taken_ms = diff.tv_sec * 1000 + (diff.tv_nsec / 1000000);
-
-        // ParodusInfo("nopoll_loop_wait() time %d msec\n", time_taken_ms);
+	if(time_taken_ms > 7000)
```

Comment:
It might be better to log for cases where time taken is not 5xxx as observed a case with 2xxx ms
Also log the nopoll_loop_wait() func call int return value

Reply:
Ok,made the changes as mentioned

Reply:
Do like if(time_taken_ms/1000 != 5)

Reply:
Made the change